x/wasm/keeper/snapshotter.go

```diff
@@ -99,6 +99,9 @@ func (ws *WasmSnapshotter) Restore(
 }
 
 func restoreV1(ctx sdk.Context, k *Keeper, compressedCode []byte) error {
+	if !ioutils.IsGzip(compressedCode) {
+		return types.ErrInvalid.Wrap("not a gzip")
+	}
```

Comment:
Before `ioutils.Uncompress` accepted uncompressed inputs. This diff makes gzip mandatory. Is this intended?

Reply:
yes as in L80 the snapshotter writes only gzipped data to the stream. Any non gzipped data is not from us and should be rejected.